src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/ImmutableDictionaryOfStringTValueConverter.cs

```diff
@@ -44,8 +41,7 @@ protected internal override bool OnWriteResume(Utf8JsonWriter writer, TCollectio
             }
             else
             {
-                Debug.Assert(state.Current.CollectionEnumerator is Dictionary<string, TValue>.Enumerator);
-                enumerator = (Dictionary<string, TValue>.Enumerator)state.Current.CollectionEnumerator;
+                enumerator = (IEnumerator<KeyValuePair<string, TValue>>)state.Current.CollectionEnumerator;
```

Comment:
What's the reason for this change?

Reply:
The initial code is incorrect, as the enumerator for these collections can't be cast to `Dictionary<string, TValue>.Enumerator`. We didn't have coverage for this scenario (resuming after fetching more data during async deserialization of immutable dictionaries) prior to this PR. We'd have leaked an `InvalidCastException`.